service/src/main/java/bio/terra/workspace/amalgam/landingzone/azure/LandingZoneApiDispatch.java

```diff
@@ -120,6 +133,19 @@ public ApiAzureLandingZoneResourcesList listAzureLandingZoneResources(String lan
     return result;
   }
 
+  public List<ApiAzureLandingZoneDeployedResource> listSubnetsWithParentVNetByPurpose(
+      String landingZoneId, LandingZonePurpose purpose) {
+
+    if (StringUtils.isBlank(landingZoneId)) {
+      throw new LandingZoneInvalidInputException("The landing zone id can't be null or empty");
+    }
+
+    return listAzureLandingZoneResources(landingZoneId).getResources().stream()
```

Comment:
Should something in here filter for subnet resource type (or am I misunderstanding)?